### ISSUE TITLE
Table - Back end

### ISSUE THREAD

**Issue by iamskip** [GemeenteNijmegen/component-library#247]

**Comment by MariannevanLierop** [comment]
@vladimirdacosta , mobiele weergave table willen Michel en Marijn even over overleggen. Of wat in Figma staat nu gewenst is. Want er zijn vragen over of dit goed gaat werken.

**Comment by MariannevanLierop** [comment]
@Michel-Verhoeven, kan gemaakt worden. 'Sorteren' hoeft niet gebouwd te worden. 
@vladimirdacosta , zou jij een nieuw ontwerp voor mobiele weergave willen maken zodat die ook gebouwd kan worden

**Comment by MariannevanLierop** [comment]
@vladimirdacosta heb jij een idee wanneer het nieuwe ontwerp af is? Dan weten we dat

**Comment by vladimirdacosta** [comment]
@MariannevanLierop ik verwacht het ontwerp + de tokens volgende week afgerond te hebben.

**Comment by MariannevanLierop** [comment]
@proudnerds-typo3 , zou jij aan Marijn kunnen aanleveren hoe tables gemaakt kunnen worden?

**Comment by proudnerds-typo3** [comment]
@mwbankers87 Een werkend voorbeeld is te zien via: https://whitelabel.nijmegen.nl/content-elementen/tabel/

In SCSS gebruik ik daar een soortgelijke opzet voor:

` table {
    clear: both;
    display: table;
    border-collapse: collapse;
    width: 100%;

    th,
    td {
        padding: 0.5em 1em;
        text-align: left;
        vertical-align: top;
    }

    thead th,
    tbody th {
        font-weight: bold;
        hyphens: auto;
    }

    tfoot {
        display: table-footer-group;
    }

    caption,
    figcaption {
        caption-side: bottom;
        font-weight: bold;
        text-align: left;
        padding-top: 0.5em;
    }

    @media (max-width: $screen-md) {
        &,
        & thead,
        & tbody,
        & tfoot,
        & th,
        & td,
        & tr {
            display: block;
        }

        caption {
            border: none;
            display: block;
        }

        p {
            hyphens: auto;
        }

        & thead tr {
            left: -9999px;
            top: -9999px;
            position: absolute;
        }

        tbody {
            tr {
                border-top: 1px solid var(--main-lightest);
                padding-bottom: pxToRem(5px);
                padding-top: pxToRem(5px);

                &:last-child {
                    border-bottom: 1px solid var(--main-lightest);
                }

                td {
                    border: 0;
                    margin-left: 50%;
                    padding-bottom: pxToRem(5px);
                    padding-top: pxToRem(5px);
                    position: relative;
                    width: 50%;
                    hyphens: auto;

                    &:before {
                        border-bottom: 0;
                        content: attr(data-title);
                        font-weight: 600;
                        float: left;
                        margin-left: -100%;
                        padding-bottom: 0;
                        padding-top: pxToRem(3px);
                        white-space: normal;
                        width: 100%;
                    }

                    &:after {
                        clear: both;
                        content: "";
                        display: block;
                        height: 0;
                        visibility: hidden;
                    }
                }
            }
        }

        tfoot {
            tr {
                border-bottom: 1px solid var(--main-lightest);
                display: flex;
                width: 100%;

                th,
                td {
                    border-bottom: 0;
                }

                th {
                    padding-left: 5%;
                    width: 50%;
                }

                td {
                    &:last-of-type {
                        text-align: right;
                    }
                }
            }
        }
    }
}`

**Comment by mwbankers87** [comment]
@vladimirdacosta Table heeft ook een issue met de tokens. Ik heb even naar de Json gekeken, en deze komen niet overeen met de Figma file.

En er lijken er ook een paar te missen die wel in de Fimga file staan. We kunnen dit maandag wel even bespreken.

**Comment by mwbankers87** [comment]
@MariannevanLierop @MariannevanLierop Het table component is toegevoegd aan [staging](https://wowebnl.github.io/design-system-nijmegen/?path=/docs/html-table--docs)

**Comment by MariannevanLierop** [comment]
@vladimirdacosta en @ElsGroffen, zouden jullie het table component willen testen op [staging](https://wowebnl.github.io/design-system-nijmegen/?path=/docs/html-table--docs)?

**Comment by ElsGroffen** [comment]
@MariannevanLierop @vladimirdacosta @mwbankers87  -> ziet er keurig uit. De tabel is prettig opgemaakt voor voorleessoftware door het gebruik van tabelheaders. Daarnaast wordt per tableheader ook de scope aangegeven wat heel prettig is. Met NVDA wordt keurig voor iedere cel aangegeven in welke rij de data staat en bij welke kolom dit hoort.

**Comment by proudnerds-typo3** [comment]
@Michel-Verhoeven & @mwbankers87 
Deze opzet kan ik alleen inzetten voor het TYPO3 component Table. Dit gaat niet werken voor tabellen die in de texteditor zijn aangemaakt. Daarbij speelt hetzelfde issue als de buttons.

Update: Ik zie dat er in de opzet geen rekening is gehouden met de tag <tfoot>. Voor het gemak neem ik dat wel mee als:
`<tfoot class="nijmegen-table__footer">`

**Comment by vladimirdacosta** [comment]
@proudnerds-typo3 Goed punt! @mwbankers87 kan je  de <tfoot>  toevoegen aan het component zodat dit niet handmatig gedaan hoeft te worden. We kunnen het eventueel optioneel maken met een parameter als het niet altijd nodig is.

Voor de rest ziet het goed uit!

**Comment by mwbankers87** [comment]
@proudnerds-typo3 @vladimirdacosta Klopt, er zit geen tfoot in op het moment. Maar ik zal deze nog even toevoegen.

Wat betreft tabellen uit de texteditor, dit is wat ik vorige week ook aangaf in de standup. Voor de classes is er misschien nog wel mapping toe te voegen. Echter het 'data-layer' element is wat lasiger. Deze heb ik wel nodig, anders krijg ik hem niet responsive zoals in het ontwerp.

**Comment by proudnerds-typo3** [comment]
@mwbankers87 Momenteel wordt de data-title gevuld via een javascript functie, dat is in TYPO3 momenteel de enige manier om dat erin te krijgen.

[responsiveTables.js](https://github.com/user-attachments/files/23160883/responsiveTables.js)

**Comment by mwbankers87** [comment]
@proudnerds-typo3 Dank, ik ga er naar kijken.

@vladimirdacosta Zal ik voor table footer cellen dezelfde styling aanhouden als de normale cellen?

**Comment by vladimirdacosta** [comment]
@mwbankers87 ja dat is prima!

**Comment by mwbankers87** [comment]
@MariannevanLierop Er is nu ook een versie van de table die semantic ondersteund wordt. Dus zonder classes. Deze staat onder [Supported semantic elements](https://wowebnl.github.io/design-system-nijmegen/?path=/docs/foundations-semantic-html-supported-semantic-elements--docs)

Er is nog een klein Storybook issue, waarbij de javascript niet consistent aangeroepen wordt. Hierdoor worden de titels in de responsive weergave niet getoond. Dit is alleen in Storybook. Na een refresh werkt het vaak wel.

Ik wil hier nog naar kijken of er een oplossing voor is.

**Comment by MariannevanLierop** [comment]
@mwbankers87 oplossen hoeft voor nu niet. Wil jij melding in het DS zetten

**Comment by mwbankers87** [comment]
@MariannevanLierop Ik heb de note toegevoegd aan het [Semantic table component](https://wowebnl.github.io/design-system-nijmegen/?path=/docs/foundations-semantic-html-supported-semantic-elements--docs)

**Comment by MariannevanLierop** [comment]
Thanks! Dan is ie voor de rest akkoord, toch @vladimirdacosta ?

**Comment by vladimirdacosta** [comment]
Ja, Akkoord! @MariannevanLierop

**Comment by MariannevanLierop** [comment]
@Michel-Verhoeven , deze is ook akkoord!

**Comment by Michel-Verhoeven** [comment]
@MariannevanLierop deze is gereleased

**Comment by MariannevanLierop** [comment]
@Michel-Verhoeven, vraagje:
Tabellen staan nu soms in een teksteditor component in Typo3 op nijmegen.nl. Kunnen jullie hetzelfde fixen voor de tabellen als dat jullie voor de buttons gedaan hebben in het Design System (zodat die automatisch de goede styling krijgen straks met de livegang).

**Comment by Michel-Verhoeven** [comment]
@MariannevanLierop zie https://gemeentenijmegen.github.io/design-system/?path=/docs/foundations-semantic-html-supported-semantic-elements--docs#table
